js/handlers.js

```diff
@@ -29,6 +29,15 @@ weechat.factory('handlers', ['$rootScope', '$log', 'models', 'plugins', 'notific
         new_date.setHours(0, 0, 0, 0);
         // Check if the date changed
         if (old_date.valueOf() !== new_date.valueOf()) {
+            var supports_formatting_date = false;
+            var should_be_tuesday = new Date(Date.UTC(2012,11,12,3,12,0))
+                .toLocaleDateString("en-US-u-ca-iso8601", {"weekday":"long"});
```

Comment:
Yeah, the code is wrong as it is.

The mixup was that I was using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/UTC#Description which is slightly different than just using the constructor.

As for DST, I was talking about DST in regards to a potential fix. The fix would be to just find January 1 1970 in local time, then make sure it's a Thursday. One would have to account for DST because getTimezoneOffset changes after DST.

However, an even better line of code than that is `new Date(1970, 0, 1).toLocaleDateString("en-US-u-ca-iso8601", {"weekday":"long"})` which should return Thursday. Which turns out is different than `new Date("1970-1-1")`, which constructs January 1st, 1970 in UTC time. `new Date(1970, 0, 1)` constructs it in local time.

Thoughts?